circletree/CircleSegment.jsx

```diff
@@ -75,9 +82,29 @@ var CircleSegment = React.createClass({
     }
   },
 
+  onActivate(e) {
+    this.setState({
+      status: ""
+    });
+    if (e.detail.origin === this) {
+      this.activate("primary");
+    }
+  },
+
+  activate(status) {
+    this.setState({ status });
+    if (this.props.activate) {
+      this.props.activate("active");
+    }
+  },
+
   render() {
+    var className = [
+      this.state.status,
+      this.props.leaf ? "leaf" : ""
+    ].join(' ');
     return (
-      <g>
+      <g className={className}>
```

Comment:
this class should probably not go on the entire group, but on the `<path>`, to make the CSS rules necessary to do the styling clearer